config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/Dependency.java

```diff
@@ -6,15 +6,29 @@
 
 package com.ca.apim.gateway.cagatewayconfig.beans;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
 import java.util.Objects;
 
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_EMPTY;
+
+@JsonInclude(NON_EMPTY)
 public class Dependency {
     private final String id;
+    @JsonIgnore
     private final Class<? extends GatewayEntity> type;
+    private final String name;
+    private final String entityType;
+    public Dependency(String id, Class<? extends GatewayEntity> type){
+        this(id, type, null, null);
+    }
 
-    public Dependency(String id, Class<? extends GatewayEntity> type) {
+    public Dependency(String id, Class<? extends GatewayEntity> type, String name, String entityType) {
```

Comment:
Required fields for us to persist in the file: **name** and **entityType**.
Remaining should be avoided.

Reply:
looks like it is used in many places already,  we will handle this separate PR

Reply:
What I mean is: ignore other fields being written to the policies.yml.

Reply:
done